### PR TITLE
Add MTE-2186 [v124] opt in notification fakespot test

### PR DESCRIPTION
**Pull request by dragosb01**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-2186)

## :bulb: Description
Added Accept the rejected Opt-in Notification fakespot smoke test.

Screenshot 2024-01-25 at 16 16 51

### CODE REVIEW

**Review by clarmso — APPROVED** [review]

**Review by isabelrios — APPROVED** [review]

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/FakespotTests.swift [hunk 1]
```diff
@@ -66,6 +66,33 @@ class FakespotTests: IphoneOnlyTestCase {
         mozWaitForElementToNotExist(app.buttons[AccessibilityIdentifiers.Toolbar.shoppingButton])
     }
 
+    // https://testrail.stage.mozaws.net/index.php?/cases/view/2358924
+    // Smoketest
+    func testAcceptTheRejectedOptInNotification() {
```
Comment (isabelrios) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/FakespotTests.swift:71]: This test has failed 1 out of 3 in the PR, can we try to make it more stable? 
~Tests/XCUITests/BaseTestCase.swift:128 - failed - Timed out waiting for element "Search Amazon" TextField to exist

Comment (dragosb01) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/FakespotTests.swift:71]: yea, i am planning to look into it, locally is hard to catch that corner case, but i will make it more stable
